service/telemetry/logger.go

```diff
@@ -51,3 +76,67 @@ func newSampledLogger(logger *zap.Logger, sc *LogsSamplingConfig) *zap.Logger {
 	})
 	return logger.WithOptions(opts)
 }
+
+func getRotationSinkFactory(cfg *configrotate.Config) func(u *url.URL) (zap.Sink, error) {
+	return func(u *url.URL) (zap.Sink, error) {
+		p := u.Query().Get("path")
+		writer, err := cfg.NewWriter(p)
+		if err != nil {
+			return nil, err
+		}
+		return nopSyncSink{writer}, nil
+	}
+}
+
+// lumberjack.Logger does not provide a Sync() method, which is required by zap.Sink
+// explanation: https://github.com/natefinch/lumberjack/pull/47#issuecomment-322502210
+type nopSyncSink struct {
+	io.WriteCloser
+}
+
+func (w nopSyncSink) Sync() error {
+	return nil
+}
+
+func setRotationURL(paths []string, rotationSchema string) ([]string, error) {
```

Comment:
I'm unclear on what we can and cannot assume about the inputs to this function, and what we are trying to do with them. Are the "paths" constrained to `stdout`, `stderr` and file paths?

Reply:
As explained in [the codes of zap](https://github.com/uber-go/zap/blob/bd0cf0447951b77aa98dcfc1ac19e6f58d3ee64f/writer.go#L50), by default the only schema `file` is supported, and missing schema will be considered as `file`.

So I only modify URLs without schema or with schema `file`. URLs with other schemas will be added as is. 
https://github.com/h0cheung/opentelemetry-collector/blob/233cbf14e9d2769c21d5226d00c2d956c8559ffb/service/telemetry/logger.go#L116-139